src/status_im/contexts/onboarding/events.cljs

```diff
@@ -67,6 +67,28 @@
  (fn [_ [error]]
    {:dispatch [:biometric/show-message (ex-cause error)]}))
 
+;; Navigate to next screen once onboarding animation and account creation both are complete
+;; Note: If account creation fails, user will get stuck on loading screen
+;; TODO: Show error screen in case account creation/restoration fails
+(rf/defn account-creation-complete
+  {:events [:onboarding/account-creation-complete]}
+  [{:keys [db] :as cofx} {:keys [loading-animation-complete? loading-screen login-signal-received?]}]
+  (let [loading-animation-complete? (or loading-animation-complete?
```

Comment:
We had a few discussions in the past about avoiding shadowing, but I'd need to dig to find them. But the decision was to prefer different names. Destructuring is what's causing the problem in this case. So an alternative that fixes shadowing:

```clojure
(rf/reg-event-fx :onboarding/account-creation-complete
 (fn [{:keys [db]} {:keys [opts]}]
   (let [loading-animation-complete? (or (:loading-animation-complete? opts)
                                         (:onboarding/loading-animation-complete? db))
         loading-screen              (or (:loading-screen opts) (:onboarding/loading-screen db))
         login-signal-received?      (or (:login-signal-received? opts)
                                         (:onboarding/login-signal-received? db))]
     (if (and loading-animation-complete? login-signal-received?)
       {:db (-> db
                (dissoc :onboarding/loading-screen)
                (dissoc :onboarding/loading-animation-complete?)
                (dissoc :onboarding/login-signal-received?))
        :fx [[:dispatch
              [:navigate-to-within-stack [:screen/onboarding.enable-notifications loading-screen]]]]}
       {:db (-> db
                (assoc :onboarding/loading-screen loading-screen)
                (assoc :onboarding/loading-animation-complete? loading-animation-complete?)
                (assoc :onboarding/login-signal-received? login-signal-received?))}))))
```